src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -923,8 +929,38 @@ let cmdline_term ~allow_empty_config : conf Term.t =
                 code_config;
                 secrets_config;
                 supply_chain_config;
+                git_remote = None;
               }
     in
+    let engine_type, target_roots =
+      let get_git_remote_config analysis =
+        let url = List.hd target_roots in
+        Logs.app (fun m ->
+            m
+              "Note that PRO interfile is turned off for remote git \
+               repositories");
+        Logs.app (fun m ->
+            m "Using OSS engine with remote git repository %s" url);
+        let git_remote = Some url in
+        Engine_type.PRO
+          {
+            extra_languages = false;
+            analysis;
+            code_config = None;
+            secrets_config = None;
+            supply_chain_config = None;
+            git_remote;
```

Comment:
Like we discussed, I think ultimately we would be best served by having this be folded in with target_roots or similar in the current `Core_scan_config.t`. If we want to allow this for just query console purposes and commit to improving it later, I think that's okay but I am apprehensive about splitting up the logic of how targets are produced into many different locations and would rather see it tied together through having all target sources unified.

Really, I think this should be added to `target_source` and `Core_scan.targets_of_config` updated to include the relevant logic for the shallow clone (plus set up whatever is needed for the sparse checkout add). There is some language related logic there but it seems stale (see, e.g., `lang` always being `None` via `Core_runner.core_scan_config_of_conf`). So then the issue is that currently returns `In.target`s and the reporting associated with `Xtarget.t`s. I don't see any actual reason that this needs to be the case. It seems like we just need to align this with whatever `Core_scan.iter_targets_and_get_matches_and_exn_to_errors` manipulates -- so if we had a type representing a target and reporting location, for instance, we could just pass that through directly and have:
- `targets_of_config` generate a list of such values
- `mk_target_handler` handle such values

In fact, this is seemingly what `Xtarget.t` is (where the reporting location is an assumed file path), although this is complicated by the fact that we send targets to subprocesses, so we would (I assume?) have issues with using `Xtarget.t` since it has lazy fields.

It would seem reasonable to me to have a sum for this, which is say, `File | GitBlob | ...` with essentially the same information as `Xtarget.t` but without lazy fields, and then have `t -> Xtarget.t` implemented for that type, with the `file` field in `Xtarget.t` being relaxed to a more general source, e.g., a string or `File of Fpath.t | Git of Fpath.t * object_sha | MetavariablePattern of Fpath.t * { starting_offset: int } | ...`.

Reply:
As one other data point of why this would be useful, the current state of this means we create some fairly silly `Xtarget.t`s in `Metavariable_pattern.get_nested_metavar_pattern_bindings` and for extract mode (where we write tempfiles we don't really fundamentally need for the "separate" case --- `Match_extract_mode.mk_extract_target`)